app/utils/communication/notifications.py

```diff
@@ -50,22 +51,37 @@ async def _manage_firebase_batch_response(
         Manage the response of a firebase notification. We need to assume that tokens that failed to be send are not valid anymore and delete them from the database.
         """
         if response.failure_count > 0:
-            responses = response.responses
-            failed_tokens = []
+            responses: list[messaging.SendResponse] = response.responses
+            failed_tokens: list[str] = []
+            mismatching_tokens: list[str] = []
             for idx, resp in enumerate(responses):
                 if not resp.success:
                     # Firebase may return different errors: https://firebase.google.com/docs/reference/admin/python/firebase_admin.messaging#exceptions
                     # UnregisteredError happens when the token is not valid anymore, and should thus be removed from the database
                     # Other errors may happen, we want to log them as they may indicate a problem with the firebase configuration
                     if not isinstance(
                         resp.exception,
-                        firebase_admin.messaging.UnregisteredError,
+                        messaging.UnregisteredError,
                     ):
-                        hyperion_error_logger.error(
-                            f"Firebase: Failed to send firebase notification to token {tokens[idx]}: {resp.exception}",
-                        )
+                        if isinstance(
+                            resp.exception,
+                            messaging.SenderIdMismatchError,
+                        ):
+                            mismatching_tokens.append(tokens[idx])
+                        else:
+                            hyperion_error_logger.error(
+                                f"Firebase: Failed to send firebase notification to token {tokens[idx]}: {resp.exception}",
+                            )
                     # The order of responses corresponds to the order of the registration tokens.
                     failed_tokens.append(tokens[idx])
+            hyperion_error_logger.error(
+                # chr(10) == "\n"
+                f"""
+                    Firebase: SenderId mismatch for users:
+                    {chr(10).join(await cruds_notification.get_user_ids_by_firebase_tokens(tokens=mismatching_tokens, db=db))}
+                """,
+            )
+            # TODO: ask to register the device again and retry sending, using the message_content arg
```

Comment:
The mobile application is responsible to registering a new token before its expiration, and in any cases at least once a month